### PR TITLE
[KYUUBI #1083] Correct wrong build strategy matrix

### PR DESCRIPTION
**Pull request by timothy65535**
### _Why are the changes needed?_

Fix wrong build strategy matrix:
spark 3.0 and spark 3.1 matrix are confused.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -36,10 +36,11 @@ jobs:
           - java: 8
             spark: '3.0'
             spark-hadoop: '2.7'
+            profiles: '-Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.0.3 -Dspark.archive.name=spark-3.0.3-bin-hadoop2.7.tgz -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.IcebergTest'
           - java: 8
             spark: '3.1'
             spark-hadoop: '2.7'
-            profiles: '-Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.0.3 -Dspark.archive.name=spark-3.0.3-bin-hadoop2.7.tgz -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.IcebergTest'
```
Comment (pan3793) [inline .github/workflows/master.yml:42]: It's by design, we use this matrix to test Kyuubi can work cross Spark versions

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -36,10 +36,11 @@ jobs:
           - java: 8
             spark: '3.0'
             spark-hadoop: '2.7'
+            profiles: '-Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.0.3 -Dspark.archive.name=spark-3.0.3-bin-hadoop2.7.tgz -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.IcebergTest'
```
Comment (pan3793) [inline .github/workflows/master.yml:39]: I think it's unnecessary, we have a profile named `spark-3.0` which will activate specific properties.

File: .github/workflows/master.yml (hunk 1 above)
Comment (timothy65535) [reply .github/workflows/master.yml:42]: Got, how about add some comments here?

File: .github/workflows/master.yml (hunk 2 above)
Comment (timothy65535) [reply .github/workflows/master.yml:39]: Thanks, checked the pom.xml

File: .github/workflows/master.yml (hunk 1 above)
Comment (timothy65535) [reply .github/workflows/master.yml:42]: Close it for design, thanks for remind me. It confused me for a long time.

**Comment by timothy65535** [comment]
Close it for design.

**Pull request closed without merging**